### PR TITLE
feat: 增加 linkFields 参数传入类型

### PR DESCRIPTION
**Pull request by GaoFuhong**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/cell/row-cell.ts [hunk 1]
```diff
@@ -229,9 +229,12 @@ export class RowCell extends HeaderCell {
   protected drawLinkFieldShape() {
     const { linkFields = [] } = this.headerConfig;
     const { linkTextFill } = this.getTextStyle();
-    const isLinkField = linkFields.some(
-      (field) => field === this.meta.key || field === this.meta.id,
-    );
+    const isLinkField =
```
Comment (wjgogogo) [inline packages/s2-core/src/cell/row-cell.ts:232]: 这一段和下面那一段一样的，抽一个方法吧

**Comment by wjgogogo** [comment]
官方实例根据增加一个自定义树的 行头链接 demo

File: packages/s2-core/src/cell/row-cell.ts (hunk 1 above)
Comment (GaoFuhong) [reply packages/s2-core/src/cell/row-cell.ts]: done.

File: packages/s2-core/src/cell/row-cell.ts [hunk 2]
```diff
@@ -229,9 +230,7 @@ export class RowCell extends HeaderCell {
   protected drawLinkFieldShape() {
     const { linkFields = [] } = this.headerConfig;
     const { linkTextFill } = this.getTextStyle();
-    const isLinkField = linkFields.some(
-      (field) => field === this.meta.key || field === this.meta.id,
-    );
+    const isLinkField = checkIfLinkField(linkFields, this.meta);
```
Comment (wjgogogo) [inline packages/s2-core/src/cell/row-cell.ts:233]:
```suggestion
    const isLinkField = checkIsLinkField(linkFields, this.meta);
```

**Review by wjgogogo — APPROVED** [review]

**Pull request merged**